gensim/corpora/_mmreader.pyx

```diff
@@ -188,7 +188,7 @@ cdef class MmReader(object):
         if offset == -1:
             return []
         if isinstance(self.input, string_types):
-            fin, close_fin = utils.smart_open(self.input), True
+            fin, close_fin = utils.open(self.input), True
```

Comment:
This doesn't seem compatible because the old default mode was `rb`. See the [Migration guide](https://github.com/RaRe-Technologies/smart_open#migrating-to-the-new-open-function).

(here and everywhere else)